v23.1/performance-recipes.md

```diff
@@ -77,15 +84,109 @@ This section provides solutions for common performance issues in your applicatio
+1. Click its **Session ID** to open the **Session Details** page.
+  <img src="{{ 'images/v23.1/ui-sessions-details-page.png' | relative_url }}" alt="Sessions Details Page" style="border:1px solid #eee;max-width:100%" />
+1. Click **Cancel Statement** to cancel the **Most Recent Statement** and thus the transaction, or click **Cancel Session** to cancel the session issuing the transaction.
+
+##### Identify transactions and objects that experienced contention
+
+To identify transactions that experienced [contention](performance-best-practices-overview.html#transaction-contention) in the past:
+
+- In the **Transaction Executions** view on the **Insights** page ([{{ site.data.products.db }} Console](../cockroachcloud/insights-page.html#transaction-executions-view) and [DB Console](ui-insights-page.html#transaction-executions-view), look for a transaction with the **High Contention** insight. Click the transaction's execution ID and view the transaction execution details, including the details of the blocking transaction.
+- Visit the **Transactions** page ([{{ site.data.products.db }} Console](../cockroachcloud/transactions-page.html) and [DB Console](ui-transactions-page.html)) and sort transactions by **Contention Time**.
+
+To view tables and indexes that experienced [contention](performance-best-practices-overview.html#transaction-contention):
+
+- Query the [`crdb_internal.transaction_contention_events`](crdb-internal.html#transaction_contention_events) table to view [transactions that have blocked other transactions](crdb-internal.html#transaction-contention-example).
+- Query the [`crdb_internal.cluster_contended_tables`](crdb-internal.html#cluster_contended_tables) table to [view all tables that have experienced contention](crdb-internal.html#view-all-tables-that-have-experienced-contention).
+- Query the [`crdb_internal.cluster_contended_indexes`](crdb-internal.html#cluster_contended_indexes) table to [view all indexes that have experienced contention](crdb-internal.html#view-all-indexes-that-have-experienced-contention).
+- Query the [`crdb_internal.cluster_contention_events`](crdb-internal.html#cluster_contention_events) table 
+to [view the tables, indexes, and transactions with the most time under contention](crdb-internal.html#view-the-tables-indexes-with-the-most-time-under-contention).
+
+##### Reduce transaction contention
+
+Because [contention](performance-best-practices-overview.html#transaction-contention) affects your cluster performance, and is often [reported after it has already resolved](#contention-events), preventing contention is highly effective:
+
+{% include {{ page.version.version }}/performance/reduce-contention.md %}
+
+### Hot spots
+
+[Hot spots](performance-best-practices-overview.html#hot-spots) are a symptom of *resource contention* and can create problems as requests increase, including excessive [transaction contention](#transaction-contention).
```

Comment:
A common high-level signal of a hot spot are CPU spikes as well. Can we call this out?

Typical flow is: 1) High CPU spikes 2) View Hot Ranges 3) View Table and index of the hot range 4) View schema of the table 5) Evaluate hash-sharded indexes

Reply:
I've added an indicator for CPU spikes, but I think I need to save the updated hot spots resolution for a separate ticket. Let me know if what I've added is OK.